src/main/java/frc/robot/commands/ArmToGoalCommand.java

```diff
@@ -63,17 +72,17 @@ public void execute() {
                 yPidController.calculate(currentPosition.getY(), goalPosition.getY())
             );
         }
+        
         arm.setGoal(updatedPosition);
     }
 
     @Override
     public boolean isFinished() {
         if((flags & FINISH_INSTANTLY) == FINISH_INSTANTLY)
             return true;
-
+        
         Translation2d pos = arm.getWristRobotSpacePosition();
         double distanceFromGoal = pos.minus(goalPosition).getNorm();
-        
-        return Math.abs(distanceFromGoal) <= DISTANCE_TOLERANCE; // distance should already be absolute, but doesn't hurt to check
+        return (Math.abs(distanceFromGoal) <= DISTANCE_TOLERANCE || ((flags & IGNORE_POSITION) == 0)) && arm.getWrist().atSetpoint(); // distance should already be absolute, but doesn't hurt to check
```

Comment:
Only confirm is wrist is atSetpoint if the setpoint above is NaN (otherwise we'll be checking for wrist even if we didn't set it, which is outside the scope of the command's function)